WalletWasabi.Tests/UnitTests/BlockNotifierTests.cs

```diff
@@ -99,6 +126,17 @@ public async Task NotifyBlocksAsync()
 			await notifier.StopAsync(CancellationToken.None);
 		}
 
+		private async Task WaitForCancelSignalAsync(TimeSpan timeSpan, CancellationToken token)
+		{
+			try
+			{
+				await Task.Delay(timeSpan, token).ConfigureAwait(false);
+			}
+			catch (TaskCanceledException)
+			{
+			}
```

Comment:
This does not feel right. Try with something like this:

```c#
var tcs = new TaskCompletionSource<bool>();
token.Register(s => ((TaskCompletionSource<bool>)s).SetResult(true), tcs);
var cancelTask = tcs.Task;
var delayTask = Task.Delay(timeSpan);
await Task.WhenAny(delayTask, cancelTask).ConfigureAwait(false);
```

You could simplify it by extracting the code that waits for the cancellation to its own method, extension method or even better make a cancellation token awaitable.

Reply:
What about this: https://github.com/zkSNACKs/WalletWasabi/pull/3811/commits/595a91708fe7d054e42c299334bb2c302e22b63d#diff-108895b8e41323112609b6f9b8b250e0R114 ?

I have started with a refactoring of this to create a generally usable class out of this. But it's better suited for a new PR.

Reply:
Noo, it is ugly. I don't want to stop this test because I think it is good but please consider something like this idea (it is what I did but much cleaner): https://github.com/dotnet/runtime/issues/14991